dms/controllers/portal.py

```diff
@@ -0,0 +1,189 @@
+# Copyright 2020 Tecnativa - Víctor Martínez
+import base64
+import logging
+
+from odoo import _, http
+from odoo.exceptions import AccessError, MissingError
+from odoo.http import request
+from odoo.osv.expression import OR
+
+from odoo.addons.portal.controllers.portal import CustomerPortal
+from odoo.addons.web.controllers.main import content_disposition, ensure_db
+
+_logger = logging.getLogger(__name__)
+
+
+class CustomerPortal(CustomerPortal):
+    def _dms_check_access(self, model, res_id, access_token=None):
```

Comment:
This doesn't contain latest change about this part AFAIK.

Reply:
In this PR appear all the changes related to FIX share option https://github.com/OCA/dms/pull/26 
(https://github.com/OCA/dms/pull/22 + https://github.com/OCA/dms/pull/26)